zrml/simple-disputes/src/lib.rs

```diff
@@ -23,65 +23,140 @@ extern crate alloc;
+            ReserveIdentifier = [u8; 8],
+        >;
+
         /// Event
         type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
 
+        /// The base amount of currency that must be bonded in order to create a dispute.
+        #[pallet::constant]
+        type DisputeBond: Get<BalanceOf<Self>>;
+
         type DisputeResolution: DisputeResolutionApi<
             AccountId = Self::AccountId,
             BlockNumber = Self::BlockNumber,
             MarketId = MarketIdOf<Self>,
             Moment = MomentOf<Self>,
         >;
 
+        /// The additional amount of currency that must be bonded when creating a subsequent
+        /// dispute.
+        #[pallet::constant]
+        type DisputeFactor: Get<BalanceOf<Self>>;
+
+        /// See [`GlobalDisputesPalletApi`].
+        #[cfg(feature = "with-global-disputes")]
+        type GlobalDisputes: GlobalDisputesPalletApi<
+            MarketIdOf<Self>,
+            Self::AccountId,
+            BalanceOf<Self>,
+        >;
+
         /// The identifier of individual markets.
         type MarketCommons: MarketCommonsPalletApi<
             AccountId = Self::AccountId,
             BlockNumber = Self::BlockNumber,
         >;
 
+        /// The maximum number of disputes allowed on any single market.
+        #[pallet::constant]
+        type MaxDisputes: Get<u32>;
+
         /// The pallet identifier.
         #[pallet::constant]
         type PalletId: Get<PalletId>;
+
+        #[pallet::constant]
+        type PredictionMarketsPalletId: Get<PalletId>;
+
+        /// Handler for slashed funds.
+        type Slash: OnUnbalanced<NegativeImbalanceOf<Self>>;
+
+        /// Weights generated by benchmarks
+        type WeightInfo: WeightInfoZeitgeist;
+    }
+
+    type BalanceOf<T> =
+        <CurrencyOf<T> as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+    pub(crate) type CurrencyOf<T> =
+        <<T as Config>::MarketCommons as MarketCommonsPalletApi>::Currency;
+    pub(crate) type NegativeImbalanceOf<T> =
+        <CurrencyOf<T> as Currency<<T as frame_system::Config>::AccountId>>::NegativeImbalance;
+    pub(crate) type MarketIdOf<T> =
+        <<T as Config>::MarketCommons as MarketCommonsPalletApi>::MarketId;
+    pub(crate) type MomentOf<T> = <<T as Config>::MarketCommons as MarketCommonsPalletApi>::Moment;
+    pub(crate) type MarketOf<T> = Market<
+        <T as frame_system::Config>::AccountId,
+        BalanceOf<T>,
+        <T as frame_system::Config>::BlockNumber,
+        MomentOf<T>,
+    >;
+
+    #[pallet::pallet]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    /// For each market, this holds the dispute information for each dispute that's
+    /// been issued.
+    #[pallet::storage]
+    pub type Disputes<T: Config> = StorageMap<
+        _,
+        Blake2_128Concat,
+        MarketIdOf<T>,
+        BoundedVec<MarketDispute<T::AccountId, T::BlockNumber>, T::MaxDisputes>,
+        ValueQuery,
+    >;
```

Comment:
Use the migration as opportunity to save the bond amount inside the new storage item `simple-disputes -> Disputes`.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/938/commits/a67c50d7204446f859afe65d1da4e668d84b9169#diff-e049e54985089861e255bbb3db36aefd2fdf77c9c7ada07b7bec60dbd067732a